guppylang/checker/expr_checker.py

```diff
@@ -949,8 +951,11 @@ def type_check_args(
     return new_args, subst
 
 
-def check_inout_arg_place(place: Place, ctx: Context, node: PlaceNode) -> Place:
-    """Performs additional checks for borrowed place arguments.
+def check_place_assignable(
+    place: Place, ctx: Context, node: ast.expr, reason: str
+) -> Place:
```

Comment:
I generalised this function since this code was duplicated in the statement checker